js_modules/dagster-ui/packages/ui-core/src/asset-data/AssetBaseDataProvider.tsx

```diff
@@ -0,0 +1,164 @@
+import {ApolloClient, gql, useApolloClient} from '@apollo/client';
+import React from 'react';
+
+import {
+  AssetGraphLiveQuery,
+  AssetGraphLiveQueryVariables,
+} from './types/AssetBaseDataProvider.types';
+import {buildLiveDataForNode, tokenForAssetKey, tokenToAssetKey} from '../asset-graph/Utils';
+import {AssetKeyInput} from '../graphql/types';
+import {liveDataFactory} from '../live-data-provider/Factory';
+import {LiveDataThreadID} from '../live-data-provider/LiveDataThread';
+
+function init() {
+  return liveDataFactory(
+    () => {
+      return useApolloClient();
+    },
+    async (keys, client: ApolloClient<any>) => {
```

Comment:
If we ever revisit this, I think it would be nice to give these two parameters names, eg `liveDataFactory({ getClient: () => ...})`, can never quiet tell by looking at it when the second one is called exactly

Reply:
yeah agreed.